File: bank-app/app/src/main/java/com/android/bankapp/login/LoginConfigurator.java
package com.android.bankapp.login;

import com.android.bankapp.login.interactor.LoginInteractor;
import com.android.bankapp.login.presenter.LoginPresenter;
import com.android.bankapp.login.view.LoginActivity;
import com.android.bankapp.login.router.LoginRouter;

import java.lang.ref.WeakReference;

public enum LoginConfigurator {

    INSTANCE;

    public void configure(LoginActivity activity){
        LoginRouter router = new LoginRouter();
        router.activity = new WeakReference<>(activity);

        LoginPresenter presenter = new LoginPresenter();
        presenter.output = new WeakReference<>(activity);

        LoginInteractor interactor = new LoginInteractor();
        interactor.output = presenter;

        if (activity.output == null){
            activity.output = interactor;
        }
        if (activity.router == null){
            activity.router = router;
        }

    }
}

File: app/src/main/java/com/br/natanfelipe/bankapp/configurator/LoginConfigurator.kt
package com.br.natanfelipe.bankapp.configurator

import com.br.natanfelipe.bankapp.interactor.HomeInteractor
import com.br.natanfelipe.bankapp.interactor.LoginInteractor
import com.br.natanfelipe.bankapp.presenter.HomePresenter
import com.br.natanfelipe.bankapp.presenter.LoginPresenter
import com.br.natanfelipe.bankapp.router.LoginRouter
import com.br.natanfelipe.bankapp.view.home.HomeActivity
import com.br.natanfelipe.bankapp.view.login.LoginActivity
import java.lang.ref.WeakReference

enum class LoginConfigurator {
    INSTANCE;

    fun configure(activity: LoginActivity) {

        /*val router = LoginRouter()
        router.activity = WeakReference<LoginActivity>(activity)*/

        val presenter = LoginPresenter()
        presenter.output = WeakReference(activity)

        val interactor = LoginInteractor()
        interactor.output = presenter

        if (activity.output == null) {
            activity.output = interactor
        }
        /*if (activity.router == null) {
            activity.router = router
        }*/
    }
}
